src/metabase/api/dashboard.clj

```diff
@@ -784,38 +786,63 @@
   "How many results to return when chain filtering"
   1000)
 
-(mu/defn ^:private mappings->field-ids :- [:maybe [:set ms/PositiveInt]]
+(defn- find-template-tag
+  "Fetch the `:field` clause from `dashcard` referenced by `template-tag`.
+
+    (find-template-tag [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
+  [dimension card]
+  (when-let [tag (first (mbql.u/get-clause :template-tag dimension))]
+    (get-in card [:dataset_query :native :template-tags (u/qualified-name tag)])))
+
+(mu/defn ^:private mappings->fields :- [:maybe [:set [:map
```

Comment:
idea: instead of taking a `mappings`, how about we make this function taking a `param`, so that it can determine the `op` as well?

with this I think `chain-filter-constraints` can greatly be simplified, thinking about the structure of that function will be

```clojure
(for [[param-key value] constraint-param-key->value
      field             (param->fields (param-key->param dashboard param-key))]
  (merge field {:value value}))
```

Reply:
Ahhh that's a great idea, plus the type becomes so much simpler.